### PR TITLE
Problem: output artifact gets expired easily

### PR DESCRIPTION
**Pull request by mmsqe**
record in log

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
	- Introduced a `Tee` class to enable simultaneous logging to both a file and the console.
	- Enhanced the logging mechanism in the `do_run` function for improved output handling.

- **Bug Fixes**
	- No significant bug fixes were reported in this release.

- **Documentation**
	- Updated documentation to reflect the new logging capabilities and usage of the `Tee` class.

### CODE REVIEW

File: testground/benchmark/benchmark/stats.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def dump_block_stats(
         blocks.append((txs, timestamp))
         tps = calculate_tps(blocks[-TPS_WINDOW:])
         tps_list.append(tps)
+        print("block", i, txs, timestamp, tps)
```
Comment (yihuang) [inline testground/benchmark/benchmark/stats.py:60]: can we search old logs from the dashboard?

Comment (mmsqe) [reply testground/benchmark/benchmark/stats.py:60]: I think so, and log is kept much longer

Comment (yihuang) [reply testground/benchmark/benchmark/stats.py:60]: another issue is the `fp` passed can be stdout already, in that case we'll duplicate it.

**Comment by mmsqe** [comment]
> I did a refactoring, please review that.

nice, can tee 2 outputs at the same time

**Pull request merged**